newIDE/app/src/ObjectEditor/Editors/SpriteEditor/SpritesList.js

```diff
@@ -230,10 +241,11 @@ export default class SpritesList extends Component<Props, void> {
           direction.getTimeBetweenFrames() > 0
             ? 1 / direction.getTimeBetweenFrames()
             : 1,
-        name: animationName ? `${animationName}` : `${objectName}`,
+        name: animationName ? `${animationName}` : `${(resourceNames.length > 0) ? path.basename(resourceNames[0], path.extname(resourceNames[0])) : objectName}`,

```

Comment:
I don't know why it was so complicated before, surely I missed that, you can remove the string interpolation:

```
name: animationName ? animationName : (resourceNames.length > 0) ? path.basename(resourceNames[0], path.extname(resourceNames[0])) : objectName,
```

or slightly shorter:

```
name: animationName || ((resourceNames.length > 0) ? path.basename(resourceNames[0], path.extname(resourceNames[0])) : objectName),
```